pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -207,6 +208,19 @@ func toMachineTemplate(machinePoolName string, cluster *rancherv1.Cluster, machi
 	return ustr, nil
 }
 
+func populateHostnameLengthLimitAnnotation(mp rancherv1.RKEMachinePool, defaults rancherv1.RKEMachinePoolDefaults, annotations map[string]string) {
+	hostnameLimit := 0
+	if hl := mp.HostnameLengthLimit; hl >= rke2.MinimumHostnameLengthLimit && hl <= rke2.MaximumHostnameLengthLimit {
+		hostnameLimit = hl
+	} else if hl = defaults.HostnameLengthLimit; hl >= rke2.MinimumHostnameLengthLimit && hl <= rke2.MaximumHostnameLengthLimit {
+		hostnameLimit = hl
+	}
+
+	if hostnameLimit != 0 {
+		annotations[rke2.HostnameLengthLimitAnnotation] = strconv.Itoa(hostnameLimit)
```

Comment:
also a nit check but this func should really have a nil check for the map

Reply:
like I know where it's used right now always instantiates the map but if its used somewhere else then it can panic